pkg/util/logging_http_transport.go

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2018 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"os"
+)
+
+type LoggingHttpTransport struct {
+	transport http.RoundTripper
+	stream    io.Writer
+}
+
+func NewLoggingTransport(transport http.RoundTripper) http.RoundTripper {
+	return &LoggingHttpTransport{transport, nil}
+}
+
+func NewLoggingTransportWithStream(transport http.RoundTripper, s io.Writer) http.RoundTripper {
+	return &LoggingHttpTransport{transport, s}
+}
+
+var SENSITIVE_HEADERS = map[string]bool{
```

Comment:
To use an empty struct instead of a boolean for saving cost (bytes) is my taste.

```golang
var SENSITIVE_HEADERS = map[string]struct{}{
	"Authorization":       {},
	"WWW-Authenticate":    {},
	"Cookie":              {},
	"Proxy-Authorization": {},
}

...

	for k, v := range r.Header {
		_, sensitive := SENSITIVE_HEADERS[k]
		if sensitive {
...


```

Reply:
I think the boolean is clearer, personally.

Reply:
... and I checked, it's the idiom https://blog.golang.org/go-maps-in-action